DefaultConfigsExtension/src/main/java/de/viadee/xai/anchor/adapter/tabular/discretizer/impl/AmevaDiscretizer.java

```diff
@@ -0,0 +1,154 @@
+import de.viadee.xai.anchor.adapter.tabular.discretizer.DiscretizationTransition;
+import de.viadee.xai.anchor.adapter.tabular.discretizer.Interval;
+
+import java.io.Serializable;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * Implementation of the Ameva discretization algorithm described by [Gonzales-Abril, Cuberos, Velasco, Ortega 2009]
+ */
+public class AmevaDiscretizer extends AbstractDiscretizer {
+
+    private Double[] targetValues;
+    private long[] targetValueDistribution;
+    private List<Double> bCutPoints = new ArrayList<>();
+    private List<Double> actualCutPoints = new ArrayList<>();
+    private List<AbstractMap.SimpleImmutableEntry<Number, Double>> keyValuePairs;
+
+    /**
+     * Constructs the Ameva Discretizer, Ameva works without any Parameters.
+     */
+    public AmevaDiscretizer() {
+        super(true);
+    }
+
+    /**
+     *
+     * Implementation of Ameva, 1. sort values and get all CutPoints, 2. Consecutively add new cut points and determine
+     * Ameva value (variation of contingency coefficient) of the tentative cut points. Get best cut point of this
+     * iteration. If the ameva value does not increase with new cut points break and return Discretization
+     *
+     * @param labels Array of Doubles, classifications of instances
+     * @param values Array of Numbers expected. Ameva is only possible with continuous variables
+     * @return list of DiscretizationTransition determined to have the highest Ameva value
+     */
+    @Override
+    protected List<DiscretizationTransition> fitCreateTransitions(Serializable[] values, Double[] labels) {
+
+        if (Stream.of(values).anyMatch(v -> !(v instanceof Number))) {
+            throw new IllegalArgumentException("Only numeric values allowed for this discretizer");
+        }
+        targetValues = Arrays.stream(labels).sorted().distinct().toArray(Double[]::new);
+        targetValueDistribution = new long[targetValues.length];
+        for(int i = 0; i < targetValues.length; i++) {
+            int finalI = i;
+            targetValueDistribution[i] = Arrays.stream(labels).filter(label -> label.equals(targetValues[finalI])).count();
+        }
+
+        keyValuePairs = IntStream.range(0, values.length)
+                .mapToObj(i -> new AbstractMap.SimpleImmutableEntry<>((Number) values[i], labels[i]))
+                .sorted(Comparator.comparing(entry -> entry.getKey().doubleValue()))
+                .collect(Collectors.toList());
+
+        bCutPoints = IntStream.range(1, values.length)
+                .mapToObj(i -> ((Double) values[i] + (Double) values[i-1] )/ 2)
+                .sorted().distinct()
+                .collect(Collectors.toList());
+
+        double globalAmeva = 0.0;
+        double ameva = createNewCutPoint(globalAmeva);
+
+        while(ameva > globalAmeva){
```

Comment:
Is this save to do? A code comment would be appreciated.

Reply:
Added a second exit-condition, which terminites the loop, im the amount of potential cut points is zero.